xdc-plugin/xdc.cc

```diff
@@ -96,17 +96,16 @@ struct GetPorts : public Pass {
 		}
 		// TODO handle more than one port
 		port_name = args.at(1);
-		char port[128];
+		std::string port_str(port_name.size(), '\0');
 		int bit(0);
-		std::string port_signal(port_name);
-		if (sscanf(port_name.c_str(), "%[^[][%d]", port, &bit) == 2) {
-			port_signal = std::string(port);
+		if (!sscanf(port_name.c_str(), "%[^[][%d]", &port_str[0], &bit)) {
+			log_error("Couldn't find port %s\n", port_name.c_str());
 		}
 
-		RTLIL::IdString port_id(RTLIL::escape_id(port_signal));
+		RTLIL::IdString port_id(RTLIL::escape_id(port_str));
```

Comment:
You might want to call `port_str.c_str()` so that `escape_id` recomputes the string length.  I don't know of `escape_id` will blindly rely on the std::string::size or not.

Reply:
I have added `port_str.resize(strlen(port_str.c_str()));` one line above, so I guess I don't need one more `port_str.c_str()`.

Reply:
Looks fine.